account_global_discount/models/account_move.py

```diff
@@ -131,6 +132,33 @@ def _set_global_discounts_by_tax(self):
                     )
                 )
                 base = discount["base_discounted"]
+        # Check all moves with tax_ids defined to check if any not discount created
+        # (tax amount is zero and only one tax is aplly)
+        if self.line_ids.filtered("tax_ids"):
+            for move in self.line_ids.filtered("tax_ids"):
+                key = tuple(move.mapped("tax_ids").ids)
+                if key in taxes_keys:
+                    if taxes_keys[key]:
```

Comment:
These two lines can be written in one 'if' statement: `if taxes_keys.get("key"):`